source/run-command.txt

```diff
@@ -0,0 +1,181 @@
+member's role in the replica set, on a database:
+
+.. literalinclude:: /includes/usage-examples/run-command.c
+    :language: c
+    :dedent:
+    :start-after: start-execute-command
+    :end-before: end-execute-command
+
+For a full list of database commands and corresponding parameters, see the
+:ref:`Additional Information <c-addl-info-runcommand>` section.
+
+Command Options
+---------------
+
+You can specify optional command behavior for the
+``mongoc_client_command_with_opts()`` function. This function accepts a BSON
+document for the ``opts`` parameter.
+
+You can pass a BSON document that specifies the following options:
+
+- ``readConcern``
+- ``writeConcern``
+- ``sessionId``
+- ``collation``
+- ``serverId``
+
+To learn more about a command and the options that it accepts, locate the
+command and follow the link in the :manual:`Database Commands
+</reference/command/>` section of the Server manual.
+
+The following code shows how to specify a ``grantRolesToUsers`` command with a
+``majority`` write concern:
+
+.. literalinclude:: /includes/usage-examples/run-command.c
+    :language: c
+    :dedent:
+    :start-after: start-execute-command-with-options
+    :end-before: end-execute-command-with-options
+
+.. note:: Read Preference
+
+   The ``mongoc_client_command_simple()`` and
+   ``mongoc_client_command_with_opts()`` functions ignore the read preference
+   setting you might have set on your database. By default, these functions use
+   the ``primary`` read preference.
+
+   To specify a read preference other than the primary read preference, you must
+   explicity pass it as an argument. The following code demonstrates how to
+   specify a read preference and use it with the
+   ``mongoc_client_command_simple()`` function:
+
+   .. code-block:: c
+
+      read_prefs = mongoc_read_prefs_new(MONGOC_READ_SECONDARY);
+
+      command = BCON_NEW("hello", BCON_INT32(1));
+
+      retval = mongoc_client_command_simple(client, "admin", command,
+      read_prefs, NULL, &reply, &error);
+  
+   For more information on read preference options, see :manual:`Read Preference
+   </core/read-preference/>` in the Server manual.
+
+Response
+--------
+
+Each function returns either a BSON document or a cursor containing the response
+from the database after the command runs. Each database command performs a
+different function, so the response content can vary across commands. However,
+every response contains documents with the following fields:
+
+- ``<command result>``: Provides fields specific to the database command. For
+  example, ``count`` returns the ``n`` field and ``explain`` returns the
+  ``queryPlanner`` field.
+- ``ok``: Indicates whether the command succeeded (``1``) or failed (``0``).
+
+Example
+-------
+
+The following code shows how you can use the
+``mongoc_client_command_with_opts()`` function to run the
+``cloneCollectionAsCapped`` command with the ``writeConcern`` option, then run
```

Comment:
S: Divide this into two since that sentence is pretty long
```suggestion
``mongoc_client_command_with_opts()`` function to run the
``cloneCollectionAsCapped`` command with the ``writeConcern`` option. It then runs
```